src/ProgressBar/index.jsx

```diff
@@ -30,19 +30,29 @@ function ProgressBarAnnotated({
   progressHint,
   thresholdHint,
   ...props
-}) {
+}, ref) => {
+  const [direction, setDirection] = React.useState('ltr');
   const progressInfoRef = React.useRef();
   const thresholdInfoRef = React.useRef();
+  const progressAnnotatedRef = React.useRef();
+  const resolvedRef = ref || progressAnnotatedRef;
   const thresholdPercent = (threshold || 0) - (now || 0);
   const isProgressHintAfter = now < HINT_SWAP_PERCENT;
   const isThresholdHintAfter = threshold < HINT_SWAP_PERCENT;
   const progressColor = VARIANTS.includes(variant) ? variant : PROGRESS_DEFAULT_VARIANT;
   const thresholdColor = VARIANTS.includes(thresholdVariant) ? thresholdVariant : THRESHOLD_DEFAULT_VARIANT;
 
+  useEffect(() => {
+    if (resolvedRef.current) {
+      const pageDirection = window.getComputedStyle(resolvedRef.current).getPropertyValue('direction');
```

Comment:
@adamstankiewicz @viktorrusakov What do you think about creating such hook called `usePageDirection` where you can pass `ref` and retrieve `"rtl"/"ltr"` values as a result. Right now we have only one instance where it can be used. Maybe other components would also require additional changes in Javascript in order to achieve proper RTL display.

Reply:
Honestly, I'm not sure if it would be useful. Usually the page direction is set on the `body` or `html` tags so that the value is propagated to all elements on the page and there is no need to check specific element. Same here - why don't you just do
```javascript
const direction = window.getComputedStyle(document.body).getPropertyValue('direction');
```
this would allow you to remove this `useEffect` and state management.

I don't think there ever will be a case where `body`'s direction is different from this component's.

Reply:
@viktorrusakov In the DOM structure of the Paragon documentation site, we always have `dir` at the `body` tag. I checked some MFEs and noticed that the tag for indicating the direction might be different, for example in the [Auth MFE](https://app-studio-mfe-dev.raccoongang.com/authn/login?next=https%3A%2F%2Fapp-studio-mfe-dev.raccoongang.com%2Fcourse-authoring%2Fcourse%2Fcourse-v1%3AedX%2BDemoX%2BDemo_Course%2Fsettings%2Fadvanced) it is specified for `html` tag. It seems that if, in the case of the [ProgressBar component](https://paragon-openedx.netlify.app/components/progressbar/#annotated-variant), we get dir from the `body` tag, we run the risk of getting `undefined` in some of the MFEs.
After discussing with @adamstankiewicz the option of defining this functionality in the form of a custom React hook, we decided to implement it and provide css-in-JS users with an alternative to `[dir="rtl"]` from CSS.

Reply:
Why do you think you would get `undefined`? If `dir` is specified in `html` tag that just means that all of the elements on the page will inherit this value, even `body`. I just tested it locally with our docs site and my solution works even with setting `dir` to `html` tag.

I think it's okay to provide a custom hook, but it looks like an overkill for this component. It just adds unnecessary state logic and complexity.

Reply:
@viktorrusakov Ok, I agree with you
Thanks 👍